src/runtime/crdt/crdt.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export type VersionMap = Map<string, number>;
+
+// A CRDT model is parameterized by:
+//  - the operations that can be applied
+//  - the internal data representation of the model
+//  - the external (particle-facing) data representation of the model
+// A CRDT model can:
+//  - merge with other models. This produces a 2-sided delta 
+//    (change from this model to merged model, change from other model to merged model)
+//  - apply an operation. This might fail (e.g. if the operation is out-of-order), in which case
+//    applyOperation() will return false.
+//  - report on internal data
+//  - report on the particle's view of the data.
```

Comment:
These last four bullets are probably better as comments on the methods themselves.